pkg/client/unversioned/clientcmd/api/types.go

```diff
@@ -92,6 +94,8 @@ type AuthInfo struct {
 	Username string `json:"username,omitempty"`
 	// Password is the password for basic authentication to the kubernetes cluster.
 	Password string `json:"password,omitempty"`
+	// AuthProvider specifies custom authentication for the kubernetes cluster.
+	AuthProvider *AuthProviderConfig `json:"auth-provider,omitempty"`
```

Comment:
+1

On Tue, Mar 22, 2016 at 10:28 AM, David Eads <notifications@github.com>
wrote:

> In pkg/client/unversioned/clientcmd/api/types.go
> <https://github.com/kubernetes/kubernetes/pull/23066#discussion_r56994711>
> :
>
> > @@ -92,6 +94,8 @@ type AuthInfo struct {
> >  	Username string `json:"username,omitempty"`
> >  	// Password is the password for basic authentication to the kubernetes cluster.
> >  	Password string `json:"password,omitempty"`
> > +	// AuthProvider specifies custom authentication for the kubernetes cluster.
> > +	AuthProvider *AuthProviderConfig `json:"auth-provider,omitempty"`
>
> Good point. I can see this being the case for us over the next few months.
> Were you thinking that this could be a slice instead?
>
> Yes, with the godoc describing the wrapping order and warning that we
> provide no protection against one wrapper undoing the work of another
> wrapper.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/23066/files/6513f0d96e0e699f117afee7b1f59838044980e4#r56994711>
>